posthog/hogql_queries/insights/trends/trends_query_runner.py

```diff
@@ -119,9 +54,19 @@ def _refresh_frequency(self):
 
         return refresh_frequency
 
-    def to_persons_query(self) -> str:
-        # TODO: add support for selecting and filtering by breakdowns
-        raise NotImplementedError()
+    def to_query(self) -> List[ast.SelectQuery]:
```

Comment:
The parent query runner returns just one `ast.SelectQuery` in the `to_query` method. This should power the "to hogql" button next to insights in the interface. Returning multiple queries breaks that flow.

Is there some way to return one query from here, even if just a large `union all`... and the runner will keep running individual queries?

Also, one other consideration: when we move to CH Cloud, we'll have access to a lot of parallelisation. This runner will run each query serially. Not blocking for now, but maybe we'll want to move the merge and formula into clickhouse somehow.

Reply:
Would be keen to change this in an upcoming PR - can add it to my list of things TODO. It's been like this from the beginning of the trends query runner.

I'm expecting some insights won't run as a `UNION ALL` from the "to hogql" button due to the size of the query itself. Imagine 25 x of these queries in a single union all. If we want the `to_query` method to return a single query, then cool, but we'll likely need to break it down into each `SELECT` before passing them to the `execute` method

Reply:
Yep, but ClickHouse Cloud should ideally then parallelise those 25 queries and get the results much faster than us doing the loop in Python ever will.